pydantic/_internal/_model_construction.py

```diff
@@ -162,8 +163,10 @@ def wrapped_model_post_init(self: BaseModel, __context: Any) -> None:
                 obj.__set_name__(cls, name)
 
             if __pydantic_reset_parent_namespace__:
-                cls.__pydantic_parent_namespace__ = parent_frame_namespace()
+                cls.__pydantic_parent_namespace__ = build_lenient_weakvaluedict(parent_frame_namespace())
             parent_namespace = getattr(cls, '__pydantic_parent_namespace__', None)
+            if isinstance(parent_namespace, dict):
+                parent_namespace = unpack_lenient_weakvaluedict(parent_namespace)
```

Comment:
This check maybe redundant as there is already the None check inside `unpack_lenient_weakvaluedict `